lib/dictBuilder/zdict.c

```diff
@@ -99,6 +100,17 @@ unsigned ZDICT_getDictID(const void* dictBuffer, size_t dictSize)
     return MEM_readLE32((const char*)dictBuffer + 4);
 }
 
+size_t ZDICT_getDictHeaderSize(const void* dictBuffer, size_t dictSize)
+{
+    if (dictSize <= 8 || MEM_readLE32(dictBuffer) != ZSTD_MAGIC_DICTIONARY) return 0;
+
+    {   ZSTD_entropyDTables_t dummyEntropyTables;
```

Comment:
`ZSTD_entropyDTables_t` is a 28 KB structure.
If it was a `ZSTD_*` function, it would be considered too large to store it on stack.
Though as a `ZDICT_*` entry with a more narrow use case, I'm less sure. 
@terrelln, what do you think ?

Reply:
I'd prefer to allocate it in the heap. It might be okay as-is, but I'd rather be safe.